modules/cleanmedianetBidAdapter.js

```diff
@@ -0,0 +1,297 @@
+        },
+        device: {
+          ua: navigator.userAgent
+        },
+        imp: [],
+        ext: {}
+      };
+
+      if (
+        bidderRequest.gdprConsent &&
+        bidderRequest.gdprConsent.consentString &&
+        bidderRequest.gdprConsent.gdprApplies
+      ) {
+        rtbBidRequest.ext.gdpr_consent = {
+          consent_string: bidderRequest.gdprConsent.consentString,
+          consent_required: bidderRequest.gdprConsent.gdprApplies
+        };
+      }
+
+      const imp = {
+        id: transactionId,
+        instl: params.instl === 1 ? 1 : 0,
+        tagid: adUnitCode,
+        bidfloor: params.bidfloor || 0,
+        bidfloorcur: 'USD',
+        secure: 1
+      };
+
+      const hasFavoredMediaType =
+        params.favoredMediaType &&
+        this.supportedMediaTypes.includes(params.favoredMediaType);
+
+      if (!mediaTypes || mediaTypes.banner) {
+        if (!hasFavoredMediaType || params.favoredMediaType === BANNER) {
+          const bannerImp = Object.assign({}, imp, {
+            banner: {
+              w: sizes.length ? sizes[0][0] : 300,
+              h: sizes.length ? sizes[0][1] : 250,
+              pos: params.pos || 0,
+              topframe: bidderRequest.refererInfo.reachedTop
+            }
+          });
+          rtbBidRequest.imp.push(bannerImp);
+        }
+      }
+
+      if (mediaTypes && mediaTypes.video) {
+        if (!hasFavoredMediaType || params.favoredMediaType === VIDEO) {
+          let videoImp = {
+            video: {
+              protocols: params.protocols || [1, 2, 3, 4, 5, 6],
+              pos: params.pos || 0,
+              ext: {context: mediaTypes.video.context}
+            }
+          };
+
+          let playerSize = mediaTypes.video.playerSize || sizes;
+          if (utils.isArray(playerSize[0])) {
+            videoImp.video.w = playerSize[0][0];
+            videoImp.video.h = playerSize[0][1];
+          } else if (utils.isNumber(playerSize[0])) {
+            videoImp.video.w = playerSize[0];
+            videoImp.video.h = playerSize[1];
+          } else {
+            videoImp.video.w = 300;
+            videoImp.video.h = 250;
+          }
+
+          videoImp = Object.assign({}, imp, videoImp);
+          rtbBidRequest.imp.push(videoImp);
+        }
+      }
+
+      if (rtbBidRequest.imp.length === 0) {
+        return;
+      }
+
+      return {
+        method: 'POST',
+        url: rtbEndpoint,
+        data: rtbBidRequest,
+        bidRequest
+      };
+    });
+  },
+
+  interpretResponse: function (serverResponse, bidRequest) {
+    const response = serverResponse && serverResponse.body;
+    if (!response) {
+      utils.logError('empty response');
+      return [];
+    }
+
+    const bids = response.seatbid.reduce(
+      (acc, seatBid) => acc.concat(seatBid.bid),
+      []
+    );
+    let outBids = [];
+
+    bids.forEach(bid => {
+      const outBid = {
+        adId: bidRequest.bidRequest.adUnitCode,
```

Comment:
What is the reasoning for doing this? Setting an adId to the adUnitCode seems pretty unusual.

Also, seems like if there are multiple server responses all will have the same adUnitCode assigned.

Reply:
Hi @robertrmartinez,
I have removed the unnecessary properties`adId` and  `adUnitCode`.
Thanks for the insight!

In addition,
I saw the tests failure in Circle CI, but it seems that it is not related to our adaptor, can you kindly give me some information about the reason of the failure?
Thank you very much!